src/Websockettest.js

```diff
@@ -554,53 +573,73 @@ const RMBTTest = (function() {
 
     /**
      * Download n Chunks from the test server
+     * @param {RMBTTestThread} thread containing an open socket
      * @param {Number} total how many chunks to download
-     * @param {RMBTThread} thread containing an open socket
-     * @param {Callback} onsuccess expects one argument (String)
+     * @param {Number} chunkSize size of single chunk in bytes
+     * @param {Function} onsuccess expects one argument (String)
      */
     function downloadChunks(thread, total, chunkSize, onsuccess) {
         //console.log(String.format(Locale.US, "thread %d: getting %d chunk(s)", threadId, chunks));
         let socket = thread.socket;
         let remainingChunks = total;
         let expectBytes = _chunkSize * total;
         let totalRead = 0;
+        let lastBuffer = null;
+
+        // https://stackoverflow.com/questions/33702838/how-to-append-bytes-multi-bytes-and-buffer-to-arraybuffer-in-javascript
```

Comment:
Thanks for the backwards compatibility fix! 👍 
Maybe add a comment that this is just for backwards compatibility with RMBTv0.3 since I'm sure that I don't will remember why this code needs to exist :)
Did you check if the new checks have any implication on cpu usage or test speed?